pkg/types/message_envelope.go

```diff
@@ -38,6 +40,19 @@ type MessageEnvelope struct {
 // NewMessageEnvelope creates a new MessageEnvelope for the specified payload with attributes from the specified context
 func NewMessageEnvelope(payload []byte, ctx context.Context) MessageEnvelope {
 	envelope := MessageEnvelope{
+		Checksum:      "",
```

Comment:
might checksum also be assigned via context? If not set also resolves to empty string.

Reply:
Yes, that's another way to do it which would also eliminate the need for the `NewMessageEnvelopeWithChecksum` function below. Effectively the same approach as we have with CorrelationID / ContentType.

Reply:
I like the idea of having the caller simply just add the Checksum property after calling the default constructor. This makes it more deterministic.

Reply:
I added functionality for getting the checksum from the context. I also added the string constant `checksum-payload` so that the key for the context map can be consistent when using this functionality from the `edgex-go` side